yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -192,13 +241,27 @@ impl ToTokens for HtmlTag {
             let name = &listener.label.name;
             let callback = &listener.value;
 
-            quote_spanned! {name.span()=> {
-                ::yew::html::#name::Wrapper::new(
-                    <::yew::virtual_dom::VTag as ::yew::virtual_dom::Transformer<_, _>>::transform(
-                        #callback
-                    )
-                )
-            }}
+            if listener.label.optional.is_some() {
+                quote_spanned! {name.span()=> {
+                    if let Some(__yew_callback) = #callback {
+                        Some(::yew::html::#name::Wrapper::new(
+                            <::yew::virtual_dom::VTag as ::yew::virtual_dom::Transformer<_, _>>::transform(
+                                __yew_callback
+                            )
+                        ))
+                    } else {
+                        None
+                    }
```

Comment:
Can be written using `Option::map` like above.

Reply:
I actually considered using `map`, but the reason I did't is because the error message you'd get from the compiler when using the attributes incorrectly was worse in those cases.

Reply:
The error currently reads like this:
```rust
html! { <input onfocus?=Callback::from(|_| ()) /> };
               ^^^^^^^  ---------------------- this expression has type `yew::callback::Callback<_>`
               |
               expected enum `yew::callback::Callback`, found enum `std::option::Option`
```

With my suggested changes it looks like this:
```rust
html! { <input onfocus?=Callback::from(|_| ()) /> };
                        ^^^^^^^^^^^^^^^^^^^^^^
                        |
                        expected enum `std::option::Option`, found enum `yew::callback::Callback`
                        help: try using a variant of the expected enum: `Some(Callback::from(|_| ()))`
```

I believe this to be a huge improvement as it even manges to suggest an appropriate fix!

Here's the code I used to get this error:
```suggestion
                    {
                        ::std::option::Option::map(#callback, |__yew_callback| {
                            ::yew::html::#name::Wrapper::new(
                                <::yew::virtual_dom::VTag as ::yew::virtual_dom::Transformer<_, _>>::transform(
                                    __yew_callback,
                                ),
                            )
                        })
                    }
```
Perhaps you forgot to use the qualified syntax in your tests?

Reply:
Btw, I usually place the braces (`{}`) in `quote!` or `quote_spanned!` on a separate line.
This is to make it clear that they're part of the generated code and not just syntax for the macros.

```rust
quote_spanned! {span=> {
  // easy to miss that the braces aren't part of the macro syntax
}}

// vs

quote_spanned! {span=>
  {
    // requires more space but it's easy to tell that this is a block.
  }
}

```